bigml/lda.py

```diff
@@ -0,0 +1,310 @@
+                                                  resource_type='cluster',
+                                                  method='get'))
+            query_string = ONLY_MODEL
+            lda_model = retrieve_resource(api, self.resource_id,
+                                          query_string=query_string)
+        else:
+            self.resource_id = get_lda_id(lda_model)
+
+        if 'object' in lda_model and isinstance(lda_model['object'], dict):
+            lda_model = lda_model['object']
+
+        if 'model' in lda_model and isinstance(lda_model['model'], dict):
+            status = get_status(lda_model)
+            if 'code' in status and status['code'] == FINISHED:
+
+                model = lda_model['model']
+
+                if 'language' in model and  model['language'] is not None:
+                    lang = model['language']
+                    if lang in CODE_TO_NAME:
+                        self._stemmer = Stemmer.Stemmer(CODE_TO_NAME[lang])
+
+                tenum = enumerate(model['termset'])
+                self._term_to_idx = {self._stem(t): i for i, t in tenum}
+
+                self._seed = model['hashed_seed']
+                self._case_sensitive = model['case_sensitive']
+                self._bigrams = model['bigrams']
+
+                self._ntopics = len(model['term_topic_assignments'][0])
+
+                self._alpha = model['alpha']
+                self._ktimesalpha = self._ntopics * self._alpha
+
+                self._uni_doc_assigns = [1] * self._ntopics
+                self._uni_daspka = self._ntopics + self._ktimesalpha
+
+                self._temp = [0] * self._ntopics
+
+                assigns = model['term_topic_assignments']
+                beta = model['beta']
+                nterms = len(self._term_to_idx)
+
+                sums = [sum(n[i] for n in assigns)
+                        for i in range(self._ntopics)]
+
+                self._phi = [[0 for _ in range(nterms)]
+                             for _ in range(self._ntopics)]
+
+                for k in range(self._ntopics):
+                    norm = sums[k] + nterms * beta
+                    for w in range(nterms):
+                        self._phi[k][w] = (assigns[w][k] + beta) / norm
+
+                ModelFields.__init__(self, model['fields'])
+            else:
+                raise Exception("The topic model isn't finished yet")
+        else:
+            raise Exception("Cannot create the LDA instance. Could not"
+                            " find the 'model' key in the resource:\n\n%s" %
+                            lda_model)
+
+    def distribution(self, input_data, by_name=True):
+        """Returns the distribution of topics given the input text.
+
+        """
+        # Checks and cleans input_data leaving the fields used in the model
+        input_data = self.filter_input_data(input_data, by_name=by_name)
+
+        # Checks that all numeric fields are present in input data
```

Comment:
> From the error message I assume that LDA needs all the text fields to be provided in the input data.

Right - the comment is wrong, I'm just trying to make sure all of the fields in the model array are present in the input data.  Am I doing it correctly?